### PR TITLE
Flashcard creation/edit page

### PR DESCRIPTION
**Pull request by Daria1911**
## Description

Adding functionality enabling users to create flashcard sets and cards. Includes forms for deck creation (title, topic, subtopic, public checkbox) and card addition (question, answer,  hint).

Added two routes "/createDeck", "/createCard" used mockData, no backend connection yet.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

© 2022 GitHub, Inc.
Terms
Privacy
Security
Status

### CODE REVIEW

File: src/pages/deck/FlashcardForm.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const handleSaveCard = () => {
+    //TODO: save card on backend
+    let updated = [...cards, flashCard];
+    setCards(updated);
+
+    //clear input after savings
+    setFlashCard(FLASHCARD_FIELDS);
+  };
+
+  const handleDelete = (index) => {
```
Comment (nikkigraybeal) [inline src/pages/deck/FlashcardForm.js:50]: Add a call to the delete flashcard route. Look in the project tracker for the route URL - it needs to include the _id of the flashcard you want to delete.

This _id is sent back in the response object when a flashcard is created.

**Review by nikkigraybeal — CHANGES_REQUESTED** [review]
Great start, Dasha! I think we can add API calls to create decks and flashcards and delete flashcards as well. 

Naming - keeping names consistent really helps for readability. See below for places where names can be adjusted.

File: src/pages/login/LoginPage.js [hunk 2]
```diff
@@ -25,15 +25,14 @@ function LoginPage() {
...
+        credentials: "include", // added this line
         body: JSON.stringify(userData),
       });
 
       const data = await response.json();
-
       if (response.ok && !response.error) {
         handleLogin(data);
-        navigate("/");
+        navigate("/create-deck");
```
Comment (nikkigraybeal) [inline src/pages/login/LoginPage.js:35]: I think we want to nav to homepage on login. I assume this was here for testing purposes though :) totally fine, just remember to remove/revert things like this before creating a PR as part of cleaning up.

**Review by nikkigraybeal — CHANGES_REQUESTED** [review]
Nice work Dasha! This is working as it should and really helps get us in a good place for the demo. 
I left comments for a few minor things to change. Let me know if you or Alejandro will be working through the review comments. 

Also, some general comments:
- run prettier on all files
- remove all unused variables and imports (this should always be done before submitting a PR)

STYLING
Flashcards Form:
- needs to be centered on page down to width of 320px 
- add top margin so there is space between navbar and form (or deck info if added)
- styling of individual flashcards needs some work - I think they should be the same width as the form, centered on the page and maybe not so tall.

Deck Form: 
- add some top margin

File: src/pages/login/LoginPage.js (hunk 2 above)
Comment (Daria1911) [reply src/pages/login/LoginPage.js:36]: Absolutely just for testing, will remove it

File: src/pages/deck/Deck.js
[the remark this answers is not in the document]
Comment (Daria1911) [reply src/pages/deck/Deck.js:1]: Changed

**Review by nikkigraybeal — APPROVED** [review]
Great work, Dasha! I adjusted some minor styles but otherwise it's ready to go!

**Pull request merged**